proto/state.pb.go

```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.27.1
-// 	protoc        v3.21.7
+// 	protoc        v3.21.9
```

Comment:
Do you really want to change the protoc version?

Reply:
Yes. protoc is not in the repository, it's part of the environment. This is not repository specific, and we must constrain (let along dictate specific) tool version without a good reason. Not touching auto generated files that ought not be in the git repo in the first place is not a good reason.

Reply:
Different version of protoc triggers a modification of protoc files, which are not modified by the user in any !

We should avoid that. Isn't it?

Reply:
No. We should not have these files in the repo in the first place. But while we must, we have to accept that  innocuous changes may be added when they are regenerated. This is the lesser evil compared to forcing people to downgrade tools in their environment for no good reason.

Reply:
Yes, if we have a Docker build environment which every one uses,  then we will not have to commit the files, which the user has not ideally modified, like in this case! For now we have to live with this, the risk I see is, if any one using a slightly different (and also older version of protoc than what you are using here), will end up submitting these files again, with no real reason,

Reply:
Well, there is no real risk here as the genes are purely in the comments. But in terms churn, then yes, this will keep happening unless everyone is on exactly same version of an external tool. Which outside a containerised build environment is just not tenable.

I also very much view the presence of these files  as a temporary thing (albeit with an undefined time limit). We _have_ to have them here at the moment as a workaround for the limitation of the go build system. However we must accept this as an inevitability. This is not the right way to do  thing, and it needs to be fixed. 

Which actually reminds me, while digging into something not directly related, one the libs I was looking it had some similar problem, and its solution was to have tests in a separate package.  I don't think was specifically due to protobufs, but working around basically the same problem of tests being unnecessarily hooked by go install.